src/prefect/engine.py

```diff
@@ -191,14 +197,22 @@ async def create_then_begin_flow_run(
         raise RuntimeError(
             f"Cannot create flow run. Failed to reach API at {client.api_url}."
         ) from connect_error
-
     state = Pending()
     if flow.should_validate_parameters:
         try:
             parameters = flow.validate_parameters(parameters)
+        except (ParameterTypeError, SignatureMismatchError) as exc:
+            validation_failure_msg = str(exc)
+            state = Failed(
+                message=str(exc),
+                data=DataDocument.encode("cloudpickle", exc),
+            )
         except Exception as exc:
+            validation_failure_msg = (
+                f"Validation of flow parameters failed with an unexpected error: {exc}"
```

Comment:
```suggestion
                f"Validation of flow parameters failed with an unexpected error: {exc!r}"
```
(ensures repr is used)